src/MonoMod.RuntimeDetour/DetourManager.Native.cs

```diff
@@ -185,17 +194,44 @@ internal sealed class ChainDelegateState {
                 return dmd.Generate();
             }
 
-            private static readonly ConditionalWeakTable<Type, MethodInfo> chainMethodCache = new();
+            private static MethodInfo GenerateRemovedStub(Type origDelType, Type nextDelType) {
+                var nextInvoke = nextDelType.GetMethod("Invoke")!;
+                Helpers.Assert(nextInvoke is not null);
+
+                using var dmd = MethodSignature.ForMethod(nextInvoke, true).CreateDmd(DebugFormatter.Format($"RemovedStub<{nextDelType}>"));
+                Helpers.Assert(dmd.Module is not null && dmd.Definition is not null);
+                var module = dmd.Module;
+
+                var il = dmd.GetILProcessor();
+
+                // instantiate a new System.InvalidOperationException and throw it
+                il.Emit(OpCodes.Ldstr, "Native detour has been removed");
+                il.Emit(OpCodes.Newobj, module.ImportReference(typeof(InvalidOperationException).GetConstructor(new Type[] { typeof(string) })));
+                il.Emit(OpCodes.Throw);
+
```

Comment:
Is there a reason to keep this separate from the removed stubs for managed methods? Could we not just pass the delegate's invoke signature to that to get a method which can be used for the target delegate?

Reply:
You're right, I'll move it into the shared DetourManager code.